### PR TITLE
Implemented O.Unique 

### PR DESCRIPTION
**Pull request by trevorsibanda**
#118

### CODE REVIEW

**Comment by lastland** [comment]
Nice work! Does it make sense to test if the column would indeed auto increase in the database? (For example, in `com.typesafe.slick.testkit.tests.InsertTest`?)

**Comment by trevorsibanda** [comment]
@lastland Let me look into that, also I just realised that the codegen still implements unique as an index. Not sure how to solve this so it uses O.Unique instead. @cvogt

**Comment by cvogt** [comment]
@trevorsibanda nice work indeed :)! Check out the test failure on travis. Can you reproduce that locally? UNIQUE as a column option seems to be supported only by some database backends. We should either provide a better error message when it is not supported or translate it to an index. See http://www.w3schools.com/sql/sql_unique.asp

@lastland It wouldn't auto increase if it is just Unique not AutoInc, right?

**Comment by lastland** [comment]
@cvogt You are right. Somehow I was thinking about `O.AutoInc`.

**Comment by cvogt** [comment]
Nice. Now it is only missing a test. Happy to help with that in a pairing session.

File: slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala [hunk 1]
```diff
@@ -54,6 +54,23 @@ class InsertTest extends AsyncTest[JdbcTestDB] {
...
+    case class ARow(email: String , id: Int = 0)
+    class A(tag: Tag) extends Table[ARow](tag , "A_UNIQUEINSERT"){
+      def id = column[Int]("id" , O.AutoInc , O.PrimaryKey)
+      def email = column[String]("email" , O.Unique)
+
+      def * = (email , id)<>(ARow.tupled , ARow.unapply )
+    }
+    val atq = TableQuery[A]
+
+    db.run(atq.schema.create)
```
Comment (trevorsibanda) [inline slick-testkit/src/main/scala/com/typesafe/slick/testkit/tests/InsertTest.scala]: Seq( atq.schema.create,
atq ++= Seq( ARow("unique@site.com") , ARow("user@site.com") )) ,
atq += ARow("unique@site.com") //test this failure
)

**Comment by trevorsibanda** [comment]
@cvogt Added test

**Comment by lastland** [comment]
LGTM. :)

**Comment by cvogt** [comment]
LGTM

**Pull request merged**